ArchitectureSpec.md

```diff
@@ -0,0 +1,74 @@
+# Vision Input Team:
+Handles image, video, and camera input and generates arrays of RGB or depth data
+## Responsibilities
+- Finding Camera parameters
+- Field of View (FoV)
+- Handle multiple cameras
+- Synchronization
+## Class structure
+4 ImageGenerator classes
+- `DepthDataGenerator (single frame depth filename, still image filename)`
+- `DepthLiveGenerator (Realsense camera port int)`
+- `ImageGenerator (single image frame filename)`
+- `VideoFileGenerator(filename)`
+- `VideoLiveGenerator(camera port input int)`
+## API
```

Comment:
Do any of the individual ImageGenerator subclasses have any additional functionality, or does this API match up across all subclasses?

Reply:
Same API across all subclasses. This enables the upper layers to use them interchangeably. The fancy name for this is polymorphism.